src/java.base/share/classes/jdk/internal/classfile/impl/ClassHierarchyImpl.java

```diff
@@ -176,7 +168,8 @@ public static final class StaticClassHierarchyResolver implements ClassHierarchy
         private final Map<ClassDesc, ClassHierarchyInfo> map;
 
         public StaticClassHierarchyResolver(Collection<ClassDesc> interfaceNames, Map<ClassDesc, ClassDesc> classToSuperClass) {
-            map = new HashMap<>(interfaceNames.size() + classToSuperClass.size());
+            map = new HashMap<>(interfaceNames.size() + classToSuperClass.size() + 1);
```

Comment:
Should use HashMap.newHashMap instead
```suggestion
            map = HashMap.newHashMap(interfaceNames.size() + classToSuperClass.size() + 1);
```